### PR TITLE
Improve Docker Compose For Development

### LINKED ISSUE #166: Create bootstrap process for docker-compose development
Currently getting up and running with docker-compose is annoying and none of us actually use it.

Reasons:

- [ ] Lack of cacheing for Rust build artifacts, every build rebuilds everything
- [ ] Difficult to get genesis data loaded in -- initially let's hardcode genesis data for static addresses, we can revisit prior to testnet release

### PR DESCRIPTION
**Pull request by zbuc**
This PR partially closes #166 by creating a new `Dockerfile.dev` used by `docker-compose.override.yaml` which has cacheing of Rust build artifacts and no longer performs a release build.

### CODE REVIEW

File: Dockerfile.dev [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+COPY pd/build.rs ./pd
+# TODO If the protobuf definitions, crypto, or wallet implementations change,
+# there will be a complete rebuild. This is maybe possible to avoid.
+COPY proto ./proto
+COPY crypto ./crypto
+COPY decaf377-fmd ./decaf377-fmd
+COPY decaf377-ka ./decaf377-ka
+COPY wallet ./wallet
+COPY .git ./.git
+# Sorry about all that mess ^, but it's worth it during development.
```
Comment (hdevalence) [inline Dockerfile.dev:29]: 👍

File: Dockerfile [hunk 2]
```diff
@@ -5,41 +5,11 @@ RUN rustup component add rustfmt
...
-# TODO If the protobuf definitions, crypto, or wallet implementations change,
-# there will be a complete rebuild. This is maybe possible to avoid.
-COPY proto ./proto
-COPY crypto ./crypto
-COPY decaf377-fmd ./decaf377-fmd
-COPY decaf377-ka ./decaf377-ka
-COPY wallet ./wallet
-COPY .git ./.git
-# Sorry about all that mess ^, but it's worth it during development.
+COPY . .
```
Comment (hdevalence) [inline Dockerfile:8]: How does this relate to the COPY commands for all the caching layers in the non-release image? (I mean, I don't understand what `COPY . .` is doing)

Comment (zbuc) [reply Dockerfile:8]: The `COPY . .` copies the entirety of the working directory on the host machine into the Docker build container, i.e. copies the full source tree in. The previous statements just copied over the `Cargo.toml` files to get the dependencies and built empty dummy `pd`/`pcli` implementations to be fast.

Comment (hdevalence) [reply Dockerfile:8]: Oh, is this running the `Dockerfile.dev` first and then loading it in here or something?

**Review by hdevalence — APPROVED** [review]
this looks great!

**Comment by zbuc** [comment]
Also worth noting: any changes to crypto/decaf/protobuf/wallet will require a full rebuild, which is suboptimal, but still an improvement over where we are today!

**Pull request merged**